src/command/graph/check.rs

```diff
@@ -0,0 +1,88 @@
+use std::path::PathBuf;
+
+use anyhow::{Context, Result};
+use serde::Serialize;
+use structopt::StructOpt;
+use prettytable::{cell, row, Table};
+
+use rover_client::query::graph::check;
+
+use crate::client::get_studio_client;
+use crate::command::RoverStdout;
+use crate::utils::parsers::{parse_graph_ref, GraphRef};
+
+#[derive(Debug, Serialize, StructOpt)]
+pub struct Check {
+    /// <NAME>@<VARIANT> of graph in Apollo Studio to validate.
+    /// @<VARIANT> may be left off, defaulting to @current
+    #[structopt(name = "GRAPH_REF", parse(try_from_str = parse_graph_ref))]
+    #[serde(skip_serializing)]
+    graph: GraphRef,
+
+    /// Name of configuration profile to use
+    #[structopt(long = "profile", default_value = "default")]
+    #[serde(skip_serializing)]
+    profile_name: String,
+
+    /// Path of .graphql/.gql schema file to push
+    #[structopt(long = "schema", short = "s")]
+    #[serde(skip_serializing)]
+    schema_path: PathBuf,
+}
+
+impl Check {
+    pub fn run(&self) -> Result<RoverStdout> {
+        let client =
+            get_studio_client(&self.profile_name).context("Failed to get studio client")?;
+        let schema = std::fs::read_to_string(&self.schema_path).with_context(|| format!("Could not read file `{}`", &self.schema_path.display()))?;
+        let res = check::run(
+            check::check_schema_query::Variables {
+                graph_id: self.graph.name.clone(),
+                variant: Some(self.graph.variant.clone()),
+                schema: Some(schema),
+            },
+            &client,
+        )
+        .context("Failed to validate schema")?;
+
+        tracing::info!(
+            "Validated schema against metrics from variant {} on graph {}",
+            &self.graph.variant,
+            &self.graph.name
+        );
+        tracing::info!(
+            "Compared {} schema changes against {} operations",
+            res.changes.len(),
+            res.number_of_checked_operations
+        );
+
+        if let Some(url) = res.target_url {
+            tracing::info!("View full details here");
+            tracing::info!("{}", url.to_string());
+        }
+
+        let mut table = Table::new();
+
+        let mut num_failures: u64 = 0;
+        table.add_row(row!["Change", "Code", "Description"]);
+        for check in res.changes {
+            let change = match check.severity {
+                check::check_schema_query::ChangeSeverity::NOTICE => "PASS",
+                check::check_schema_query::ChangeSeverity::FAILURE => {
+                    num_failures += 1;
+                    "FAIL"
+                },
+                _ => unreachable!("Unknown change severity"),
+            };
+            table.add_row(row![change, check.code, check.description]);
+        }
+
+        eprintln!("{}", table);
```

Comment:
For the case where there are 0 changes, right now, the table just prints the headings with no rows

```
+--------+------+-------------+
| Change | Code | Description |
+--------+------+-------------+
```

Can we just not print the table if there's nothing in it?

Reply:
ooh good catch